client/extensions/woocommerce/state/products/reducer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { WOOCOMMERCE_EDIT_PRODUCT } from '../action-types';
+
+export const initialState = {
+	add: {
```

Comment:
I don't think we should be storing empty defaults like this, especially when given a different UI, it would be possible to add more than one at a time.

I think `edits` should represent actual user input that hasn't been synced back to WooCommerce yet. In fact, `edits` itself probably shouldn't even exist in the state until the user makes an edit.

I'd also prefer a structure like this:
```
edits: {
  adds: [
    { <variation type to add> },
    { <another new variation type> }
  ],
  updates: [
    { id: <#>, fieldToChange: <new field value> },
    { id: <#>, anotherField: <new field value> },
  ],
  deletes: [
    <id to delete>,
    <another id to delete
  ]
}
```

And then upon the "Save" operation, you can have a single piece of code iterate this structure and make the appropriate API calls. Then, when you've confirmed it's all been done, delete the entire `edits` object from the redux state tree.